dashboard/app/models/lesson.rb

```diff
@@ -85,6 +87,18 @@ def self.add_lessons(script, lesson_group, raw_lessons, counters, new_suffix, ed
     end
   end
 
+  def self.prevent_changing_stable_i18n_key(script, raw_lesson)
+    if script.is_stable && ScriptConstants.i18n?(script.name) && !(I18n.t "data.script.name.#{script.name}.lessons").keys.map(&:to_s).include?(raw_lesson[:key])
```

Comment:
can this be `... && !(I18n.t "data.script.name.#{script.name}.lessons.#{raw_lesson[:key]})" ?

Reply:
I'd second that! There are some subtleties around the idea of retrieving non-string values from the i18n system. See https://github.com/code-dot-org/code-dot-org/pull/23745 for more context